docs/src/security.md

```diff
@@ -165,15 +165,66 @@ namespaced resources.
   validate them before starting the restore process.
 
 `nodes`
-: The operator needs to get the labels for Affinity and AntiAffinity, so it can
-  decide in which nodes a pod can be scheduled preventing the replicas to be
-  in the same node, specially if nodes are in different availability zones. This
-  permission is also used to determine if a node is schedule or not, avoiding
-  the creation of pods that cannot be created at all.
+: The operator needs to get the labels for Affinity and AntiAffinity so it can
+  decide in which nodes a pod can be scheduled. This is useful, for example, to
+  prevent the replicas from being scheduled in the same node - especially
+  important if nodes are in different availability zones. This
+  permission is also used to determine whether a node is scheduled, preventing
+  the creation of pods on unscheduled nodes,  or triggering a switchover if
+  the primary lives in an unscheduled node.
 
 To see all the permissions required by the operator, you can run `kubectl
 describe clusterrole cnpg-manager`.
 
+### ClusterRole Permissions and Operator Deployment
+
+#### Using the Manifest
+
+In the default installation for a vanilla Kubernetes deployment, the operator
+is deployed using a manifest built with `Kustomize` on the YAML files generated
+by `controller-gen` using the markers available in
+[Kubebuilder](https://book.kubebuilder.io/).
+
+Although this method is currently somewhat limited, it allows creating a
+manifest with permissions set as either `RoleBinding` or `ClusterRoleBinding`.
+
+In the default deployment, permissions are set to `ClusterRoleBinding`.
+
+#### Using OLM
+
+From a security perspective, the Operator Lifecycle Manager (OLM) provides a
+more flexible deployment method. The community offers support for building the
+necessary files and packages to deploy the operator from
+[OperatorHub.io](https://operatorhub.io/operator/cloudnative-pg). OLM allows
+users to configure the operator to watch either all namespaces, or specific
+namespaces, enabling more granular permission management.
+
+#### Why Are ClusterRole Permissions Needed?
+
+Currently, the operator requires ClusterRole permissions only to get `nodes`
+objects. All other permissions can be namespace-scoped (i.e. Role), or
+cluster-wide (i.e. ClusterRole).
+
+Even with these permissions, if someone gains access to the `ServiceAccount`,
+they will only have `get`, `list`, and `watch` permissions, which are limited
+to viewing resources. If an unauthorized user already has access to the
+`ServiceAccount`, it indicates a more significant security issue.
+
+It's crucial to prevent users from accessing the operator's `ServiceAccount`
+and any other `ServiceAccount` with elevated permissions.
+
+#### Can This Issue Be Mitigated?
```

Comment:
Don't like this section. Firstly, what "Issue" is it making reference to?
Also, it looks way too apologetic.